problem_builder/mixins.py

```diff
@@ -261,3 +264,30 @@ def delete_key(self, dictionary, key):
         except KeyError:
             pass
         return dictionary
+
+
+class MakeURLAbsoluteMixin(object):
+    url_pattren = re.compile('<img\s*src="(.*)"')
+
+    def make_url_absolute(self, text):
+        if not text:
+            return text
+
+        try:
+            from static_replace import replace_static_urls
+        except ImportError:
+            return text
+
+        lms_relative_url = replace_static_urls(text, course_id=self.course_id)
+        return self._append_lms_base(lms_relative_url)
+
+    def _append_lms_base(self, text):
+        urls = self.url_pattren.findall(text)
+        if not urls:
+            return text
+
+        lms_base = settings.ENV_TOKENS.get('LMS_BASE')
```

Comment:
XBlocks should not be importing the LMS django settings, since that goes outside of the XBlock API. What problem are we solving by using a full absolute URL instead of just the result of `replace_static_urls` ?

Reply:
@bradenmacdonald this has nothing to do with drag and drop. we're context switching to support graphic images and static links in general to appear on mobile for problem builder and step builder. if you recall from this ticket: https://edx-wiki.atlassian.net/browse/MCKIN-7725, we reached an implementation decision to fix these two xblocks so that it always returns absolute links only, without any changes from the course authoring team. So, I believe @jawad-khan had pb and sb call expand_static_url on file fields or self.runtime.replace_urls() on HTML fields before serving them via its API. so based on this, does @jawad-khan PR make sense to you?

Reply:
@janomanningham  I totally understand that, and I agree we want to add `expand_static_url` here. I just am not sure if the extra step of adding `LMS_BASE` is required here, and it goes around the XBlock API, which should generally be avoided.

The reason I mention drag and drop is that we applied a similar fix to drag and drop a long time ago, and adding the LMS_BASE part wasn't necessary then AFAIK.

Reply:
@bradenmacdonald I have used expand_static_url now(thanks for the suggestion).
I was using LMS_BASE because it was easy for mobile side to just show the data as it is. I have talked to them and they say they will make changes on their end to append base url.
I have removed LMS_BASE part now.

Reply:
@jawad-khan Thanks! Sounds great to me.